### PR TITLE
More cleanup.

### PR DESCRIPTION
**Pull request by ahmarsuhail**
- Removes deprecation warnings suppressions as those classes were deprecated as part of https://github.com/apache/hadoop/pull/4698, but are no longer deprecated with the upgrade. 
- Removes unused imports

### CODE REVIEW

File: hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3ClientFactory.java [hunk 1]
```diff
@@ -53,25 +52,14 @@
...
-   * @param uri S3A file system URI
-   * @param parameters parameter object
-   * @return S3 client
-   * @throws IOException IO problem
-   */
-  AmazonS3 createS3Client(URI uri,
-      S3ClientCreationParameters parameters) throws IOException;
 
   /**
    * Creates a new {@link S3Client}.
```
Comment (passaro) [inline hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3ClientFactory.java:69]: Shall we now rename this to `createS3Client`?

File: hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/auth/delegation/S3ADelegationTokens.java [hunk 2]
```diff
@@ -228,6 +228,9 @@ protected void serviceStop() throws Exception {
     try {
       super.serviceStop();
     } finally {
+      // TODO: Importing the transfer manager outside preview outside of the bundle causes some
```
Comment (passaro) [inline hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/auth/delegation/S3ADelegationTokens.java]: One extra "outside"

File: hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/MockS3ClientFactory.java [hunk 3]
```diff
@@ -33,10 +33,8 @@
  * An {@link S3ClientFactory} that returns Mockito mocks of the {@link AmazonS3}
  * interface suitable for unit testing.
  */
-@SuppressWarnings("deprecation")
 public class MockS3ClientFactory implements S3ClientFactory {
```
Comment (passaro) [inline hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/MockS3ClientFactory.java:36]: Maybe leave a TODO here

**Comment by passaro** [comment]
Just a couple of minor notes

File: hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3ClientFactory.java (hunk 1 above)
Comment (ahmarsuhail) [reply hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3ClientFactory.java:69]: I'm adding this method back in now, as removing it completely is slightly more complicated. I'll do that in a seperate PR.

**Review by passaro — APPROVED** [review]

**Pull request merged**